src/Features/Core/Portable/NavigationBar/IRemoteNavigationBarItemService.cs

```diff
@@ -36,32 +36,37 @@ internal sealed class SerializableNavigationBarItem
         public readonly int Indent;
         [DataMember(Order = 6)]
         public readonly ImmutableArray<SerializableNavigationBarItem> ChildItems;
-        [DataMember(Order = 7)]
-        public readonly ImmutableArray<TextSpan> Spans;
 
         // Set when kind == RoslynNavigationBarItemKind.Symbol
 
+        [DataMember(Order = 7)]
+        public readonly string? Name;
         [DataMember(Order = 8)]
-        public readonly SymbolKey? NavigationSymbolId;
-
+        public readonly bool IsObsolete;
         [DataMember(Order = 9)]
+        public readonly ImmutableArray<TextSpan> Spans;
+        [DataMember(Order = 10)]
+        public readonly TextSpan? SelectionSpan;
+        [DataMember(Order = 11)]
+        public readonly SymbolKey? NavigationSymbolId;
+        [DataMember(Order = 12)]
         public readonly int NavigationSymbolIndex;
```

Comment:
NavigationSymbolId and NavigationSymbolIndex currently exist but will be removed in my followup PR.  tools must not go back from tehse to symbols.  instead, the service should accurately and uniquely define the behavior the consuming services need through simple data it passes back.